### PR TITLE
feature(CS-2969) - submit a form from the stepper

### PR DESCRIPTION
**Pull request by jonathanweyermann**

### CODE REVIEW

File: apps/form-app/src/app/containers/Form.tsx [hunk 1]
```diff
@@ -77,6 +77,13 @@ const FormComponent: FunctionComponent<FormProps> = ({ className }) => {
                   showSubmit={showSubmit}
                   saving={busy.saving}
                   onChange={({ data, errors }) => {
+                    if (
+                      errors[0]?.message === 'should be equal to one of the allowed values' &&
+                      errors[0]?.schemaPath.includes('enum')
+                    ) {
+                      errors = null;
+                    }
+
```
Comment (rstyan) [inline apps/form-app/src/app/containers/Form.tsx:86]: This seems a bit dodgy.  Why is this code trying to filter out very specific error messages?  And even if this is a valid thing to do, the code would be much cleaner (and easier to digest) if you made this into a function, e.g.
errors = filterIgnoredErrors(errors)

File: libs/core-common/src/jsonSchema/ajv.ts [hunk 2]
```diff
@@ -21,7 +21,18 @@ export class AjvValidationService implements ValidationService {
...
+      const newSchema = JSON.parse(JSON.stringify(schema));
+
+      Object.keys(newSchema?.properties || {}).forEach((propertyName) => {
+        const property = newSchema.properties || {};
+        if (property[propertyName]?.enum?.length === 1 && property[propertyName]?.enum[0] === '')
+          delete property[propertyName].enum;
+        if (property[propertyName]?.format === 'file-urn') {
+          delete property[propertyName].format;
+        }
+      });
```
Comment (rstyan) [inline libs/core-common/src/jsonSchema/ajv.ts:33]: This code should also be turned into a function and unit tested separately.

File: libs/jsonforms-components/src/lib/Context/index.tsx [hunk 3]
```diff
@@ -89,12 +89,18 @@ interface FileManagement {
   downloadFile?: (file: File) => void;
   deleteFile?: (file: File) => void;
 }
+interface SubmitManagement {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  submitForm?: (any: any) => void;
```
Comment (rstyan) [inline libs/jsonforms-components/src/lib/Context/index.tsx:94]: are the types really any?  There are way too many places in our code where "any" can be easily replaced with an actual type.  This makes the code much, much more managable.

File: libs/jsonforms-components/src/lib/Controls/FormStepper/FormStepper.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+          scope: '#/properties/vegetarianOptions/properties/vegan',
+        },
+        {
+          type: 'Control',
+          scope: '#/properties/vegetarianOptions/properties/favoriteVegetable',
+        },
+      ],
+    },
+  ],
+};
```
Comment (rstyan) [inline libs/jsonforms-components/src/lib/Controls/FormStepper/FormStepper.spec.tsx:308]: Again, your tests do not need such elaborate schema's.  Either make them simpler or have you tests use all of the fields.  The way you have it makes it very hard to determine what exactly is being tested and requires the reader to spend too much time figuring out that most of it is unnecessary.

**Review by rstyan — CHANGES_REQUESTED** [review]

File: apps/form-app/src/app/containers/Form.tsx (hunk 1 above)
Comment (jonathanweyermann) [reply apps/form-app/src/app/containers/Form.tsx:86]: I could instead change the ui schema before it gets validated. not sure which is better. maybe that's better

Comment (rstyan) [reply apps/form-app/src/app/containers/Form.tsx:86]: Generally speaking, depending on the text of an error message eventually leads to unexpected and hard to detect runtime errors.  If you change the UI schema before had, can you avoid this situation?

Comment (jonathanweyermann) [reply apps/form-app/src/app/containers/Form.tsx:86]: yes, I already kind of do a similar thing in this code already, so it wouldn't be hard

**Review by rstyan — APPROVED** [review]

**Pull request merged**